includes/class-wc-webhook.php

```diff
@@ -306,6 +308,26 @@ protected function is_already_processed( $arg ) {
 		return false !== array_search( $arg, $this->processed, true );
 	}
 
+	/**
+	 * Should we attempt a retry on this webhook?
+	 *
+	 * Passing in the item id and the http response to allow hooks the option of bypassing this retry.
```

Comment:
Might be overthinking this, or a case of premature optimization.  These inputs might not be needed, but I would want them if I was hooking in.

Reply:
If filters should expect these arguments I don't think they should be optional. One thought might be to make `WC_Webhook::retry_delivery()` return a bool and place that in the `else` block of `WC_Webhook::log_delivery()` so we're only calling this with those arguments?